### ISSUE TITLE
Support for filtered real-time subscription

### ISSUE THREAD

**Issue by sciroad** [pocketbase/pocketbase#3625]
Pocketbase supports subscription for either a single record or an entire collection. However, if we want to subscribe to a subset of the collection, this feature is currently unavailable.

Use case: Let's say we have a collection of restaurants, and we want to subscribe to a particular user's restaurants. In this scenario, we are unable to achieve this.

Possible Solution 1: We would like to have the ability to subscribe to filtered records, allowing us to specify conditions for subscriptions. For example, using a syntax like this: `collection("restaurant").subscribe(null, filter={"owner": "user123"})`. In this case, even if we provide a null value for the first parameter, we should be able to use a filter to specify our subscription criteria.

Possible Solution 2: You can support more complex API rules for collections.

**Comment by ganigeorgiev** [comment]
`filter`, `expand`, `fields`, custom query parameters and headers support for the realtime subscription APIs is already implemented in the `develop` branch and will be available with the next v0.20.0 release (_there are no ETAs_).

If you want to give it a try, you can test the [v0.20.0-rc prerelease](https://github.com/pocketbase/pocketbase/releases/tag/v0.20.0-rc) or build the `develop` branch on your own.
Also note that in order to work this requires JS SDK v0.20.0-rc or Dart SDK v0.17.0-rc.

**Comment by sciroad** [comment]
That's awesome! Thank you, looking forward to it. 🚀

**Comment by godmar** [comment]
I, too, was looking for this feature.  If it is implemented, could the [documentation](https://pocketbase.io/docs/api-realtime/#set-subscriptions) be improved?  As of now, it doesn't list "filter" (it says: "`expand`, custom headers, etc."). It's also easy to misread the bolded *You could subscribe to a single record or to an entire collection.* as though you'd have to subscribe to the entire collection without having the option of filtering.

Perhaps an example could be added to make it clear that this is supported.

**Comment by ShivamJoker** [comment]
Hi @ganigeorgiev is this feature still in pre release?
I couldn't find autocompletion in the JS SDK.

**Comment by ganigeorgiev** [comment]
@ShivamJoker It is available with v0.20.0+. Here is an example:

```js
pb.collection("example").subscribe("*", (e) => {
   ...
}, { filter: "something > 10" })
```

**Comment by ShivamJoker** [comment]
Oh perfect. It works.
Thanks a lot.
